### ISSUE TITLE
Report failing arguments produced by Gen parameters in quickCheck

### ISSUE THREAD

**Issue by thielema** [nick8325/quickcheck#95]
I want to test a property with a limited range of values, say the characters from 'a' to 'z'. I create a custom `Gen` for this purpose and run `quickCheck` on it. Unfortunately, if a test fails the failing arguments are not reported:
~~~~
QC> QC.quickCheck $ fmap (<'f') $ QC.choose ('a','z')
*** Failed! Falsifiable (after 1 test): 
~~~~
Is it possible to add this report? If not, how about an alternative form of `quickCheck` that better supports `Gen` parameters? E.g. an alternative form might be called like this:
~~~~
QC> QC.quickCheckCustom (QC.choose ('a','z')) $ \c -> c<'f'
*** Failed! Falsifiable (after 1 test): 
't'
~~~~

**Comment by thielema** [comment]
Answering myself: I think the problem is, that `quickCheck` does not know at which level the value shall be shown. The result of a `Gen` can be modified many times using `fmap` until it is feed to the "real test". `quickCheck` cannot know where the argument preprocessing ends and where the test starts. That is, it will not be possible for `quickCheck` to report arguments provided by a `Gen`.
However, the second form makes the distinction clear and thus should work. Since you can bundle multiple many `Gen`s using `liftA2` and friends, you can handle any number of `Gen` arguments with `quickCheckCustom`.
An alternative would be a function `report :: Gen a -> Gen a` that tells `quickCheck` what to report. Outer applications should override inner applications. A call might look like this:
~~~~
QC> QC.quickCheck $ fmap (<'f') $ report $ QC.choose ('a','z')
~~~~
but I would expect that `QC.choose` already has `report` integrated.

**Comment by phadej** [comment]
Short answer (in hurry), use `forAll`:

```hs
> quickCheck $ forAll (choose ('a', 'z')) $ (< 'f')
*** Failed! Falsifiable (after 2 tests):  
'z'
```

**Comment by thielema** [comment]
Yeah, that's it! Thanks a lot!
